### ISSUE TITLE
Wrapping a model using Lambda layer corrupts performance

### ISSUE THREAD

**Issue by stigersh** [keras-team/keras-applications#168]
### Summary
In the debugging process I checked the following:
I have a keras model which I wrapped in a lambda. From some reason doing it made the optimization start from a worse accuracy point, and get stuck on that value, whereas without the lambda wrapper all is good.

Works bad:
```
class mymodel(object):
    def __init__():
        def lambda_wrapper(x):
            model = Sequential()
            #some layers like:
            cell = LSTM(100, return_sequences=True)
            layer = Bidirectional(cell)
            model.add(layer)
            return model(x)

       lambda_model_layer = Lambda(lambda x: lambda_wrapper(x))
       self.model = Sequential()
       self.model.add(lambda_model_layer)
```

Works fine: 
```
class mymodel(object):
    def __init__():
        model = Sequential()
        #some layers like:
        cell = LSTM(100, return_sequences=True)
        layer = Bidirectional(cell)
        model.add(layer)
        self.model = model
```

### Environment
- Python version: 3.6
- Keras version: 2.2.4
- Keras backend with version: tensorflow 1.14.0

**Comment by stigersh** [comment]
I think I found the problem. Lambda layers weren't designed to be used with trainable variables. Thus everything inside the lambda becomes untrainable. 
For writing a custom layer with trainable variables this link should be used: 
[writing-your-own-keras-layers](https://keras.io/layers/writing-your-own-keras-layers/)
